foist/app.py

```diff
@@ -1,120 +1,244 @@
+        return query
 
-    def get_field(self, field, search_string, t='string'):
-        base = './mets:dmdSec/*/*/*/mods:'
-        try:
-            result = self.root.find(base + search_string, ns).text
-            if t == 'string':
-                return rdflib.Literal(result.replace('"', "'"))
-            elif t == 'uri':
-                return rdflib.URIRef(result)
-        except AttributeError as e:
-            log.warning(('Error parsing ' + field +
-                         ' field for item ' + self.name))
-            return rdflib.Literal('None')
-
-    def add_text_errors(self, text_errors):
-        if self.name in text_errors:
-            errors = text_errors[self.name]
-            if (errors['PDFBox err'] != '0' or
-                (errors['No new text'] != '0' and ['No old text'] != '0') or
-                    errors['Encoded'] != '0' or errors['Hex strings'] != '0'):
-                self.metadata.add((self.s, LOCAL.no_full_text,
-                                   rdflib.Literal('True')))
-            if errors['Ligatures'] != '0':
-                self.metadata.add((self.s, LOCAL.ligature_errors,
-                                   rdflib.Literal('True')))
-            if errors['Line ends'] != '0':
-                self.metadata.add((self.s, LOCAL.line_ends,
-                                   rdflib.Literal('True')))
+    def _get_field(self, search_string, field_type='single',
+                   text_to_strip=None):
+        result = []
+        if field_type == 'degree':
+            for f in self.mets.findall(base_mets_search + search_string,
+                                       mets_namespace):
+                if f.text.startswith('Thesis'):
+                    result = f.text
+        elif field_type == 'multi':
+            for f in self.mets.findall(base_mets_search + search_string,
+                                       mets_namespace):
+                result.append(f.text)
+        elif field_type == 'notes':
+            for f in self.mets.findall(base_mets_search + search_string,
+                                       mets_namespace):
+                if not f.text.startswith('Thesis'):
+                    result.append(f.text)
+        elif field_type == 'single':
+            try:
+                result = self.mets.find(base_mets_search + search_string,
+                                        mets_namespace).text
+            except AttributeError as e:
+                result = None
+        elif field_type == 'split':
+            for f in self.mets.findall(base_mets_search + search_string,
+                                       mets_namespace):
+                if not result:
+                    result = f.text
+                else:
+                    t = f.text.lstrip(text_to_strip)
+                    result += t
+        if not result:
+            result = None
+        return result
+
+    def _get_error_value(self, error):
+        result = None
```

Comment:
Since the default return value of a function is `None`, you can simplify this to:
```python
def _get_error_value(self, error):
    if self.errors[error] == '1':
        return True
```
The same goes for the `_get_full_text_error` method below.